pkg/vault/test/authorizer/TimelockAuthorizerRoot.test.ts

```diff
@@ -0,0 +1,172 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { Contract } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import TimelockAuthorizer from '@balancer-labs/v2-helpers/src/models/authorizer/TimelockAuthorizer';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+import { ZERO_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+
+describe('TimelockAuthorizer', () => {
+  let authorizer: TimelockAuthorizer, vault: Contract;
+  let root: SignerWithAddress, grantee: SignerWithAddress;
+
+  before('setup signers', async () => {
+    [, root, grantee] = await ethers.getSigners();
+  });
+
+  const WHERE_1 = ethers.Wallet.createRandom().address;
+  const WHERE_2 = ethers.Wallet.createRandom().address;
+
+  const GENERAL_PERMISSION_SPECIFIER = TimelockAuthorizer.GENERAL_PERMISSION_SPECIFIER;
+  const EVERYWHERE = TimelockAuthorizer.EVERYWHERE;
+
+  describe('root', () => {
```

Comment:
So if I got this correctly, this is the exact same test that applies to everything related with the authorizer without the adaptor / entrypoint. Is that correct?

Reply:
Yes - I separated it because it had to initialize in a separate and weird way that interfered with the other tests unless I renamed stuff, which would have made the diff harder to read.

Reply:
:+1: to that